controllers/feeds/blockAnonymousCommentV2.js

```diff
@@ -0,0 +1,30 @@
+const BetterSocialCore = require('../../services/bettersocial');
+const ErrorResponse = require('../../utils/response/ErrorResponse');
+const SuccessResponse = require('../../utils/response/SuccessResponse');
+
+module.exports = async (req, res) => {
+  const {token, userId: selfUserId} = req;
+  const {postId, commentId, source = '', reason = [], message = ''} = req.body;
+
+  let response = await BetterSocialCore.post.blockAnonymousComment(
+    res,
+    token,
+    selfUserId,
+    postId,
+    commentId,
+    source,
+    {
+      message,
+      reason
+    }
+  );
+  if (response?.isSuccess) {
+    return SuccessResponse(
+      res,
+      null,
+      response?.message || 'This Anonymous comment has been blocked successfully'
+    );
+  }
+
+  return ErrorResponse.e500(res, response?.message || 'Error in block anonymous comment');
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new function `blockAnonymousComment` in the controller seems to be well implemented. It correctly extracts necessary information from the request, calls the service function with appropriate parameters, and handles the response properly. However, there is a potential issue with error handling. If the `BetterSocialCore.post.blockAnonymousComment` function throws an exception (for example, due to a database error), it will not be caught and handled, which could lead to unhandled promise rejection.

Consider wrapping the call to `BetterSocialCore.post.blockAnonymousComment` in a try-catch block to handle any exceptions that might occur during its execution.

```diff
module.exports = async (req, res) => {
  const {token, userId: selfUserId} = req;
  const {postId, commentId, source = '', reason = [], message = ''} = req.body;

  let response;
 try {
   response = await BetterSocialCore.post.blockAnonymousComment(
     res,
     token,
     selfUserId,
     postId,
     commentId,
     source,
     {
       message,
       reason
     }
   );
 } catch (error) {
   return ErrorResponse.e500(res, 'Error in block anonymous comment');
 }
 
 if (response?.isSuccess) {
   return SuccessResponse(
     res,
     null,
     response?.message || 'This Anonymous comment has been blocked successfully'
   );
 }

 return ErrorResponse.e500(res, response?.message || 'Error in block anonymous comment');
};
```
This way, any errors that occur during the execution of `BetterSocialCore.post.blockAnonymousComment` will be caught and handled appropriately, preventing potential unhandled promise rejections.


<!-- This is an auto-generated comment by OSS CodeRabbit -->